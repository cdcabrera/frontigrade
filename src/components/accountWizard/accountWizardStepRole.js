import React from 'react';
import PropTypes from 'prop-types';
import { Form, Grid, Icon } from 'patternfly-react';
import { connect, reduxActions } from '../../redux';
import apiTypes from '../../constants/apiConstants';
import helpers from '../../common/helpers';
import CopyField from '../copyField/copyField';
import Tooltip from '../tooltip/tooltip';

class AccountWizardStepRole extends React.Component {
  componentDidMount() {
    const { getSystemConfig } = this.props;

    getSystemConfig();
  }

  // ToDo: when form valid evaluate moving to next step onEnter key-press event
  onSubmit = event => {
    event.preventDefault();
  };

  render() {
    const { awsAccountId } = this.props;

    const popover = (
      <ul className="cloudmeter-popover-list">
        <li>Log in to AWS console</li>
        <li>Search Services to go to IAM</li>
        <li>Click Roles in the left nav</li>
        <li>
          Click the <strong>Create role</strong> button
        </li>
        <li>
          Click <strong>Another AWS account</strong>
        </li>
      </ul>
    );

    return (
      <Form onSubmit={this.onSubmit}>
        <Form.FormGroup>
          <Grid.Col sm={12}>
            <ul>
              <li>
                Create a new role in the AWS
                <a href="https://console.aws.amazon.com/iam" target="_blank" rel="noopener noreferrer">
                  Identity and Access Management
                </a>
                .
                <Tooltip delayShow={100} popover={popover}>
                  <Icon type="pf" name="info" size="large" />
                  <span className="sr-only">Steps when logging into AWS Identity and Access Management</span>
                </Tooltip>
              </li>
              <li>
                <p>
                  Paste this value into the Account ID field under <strong>Another AWS account</strong>:
                </p>
                <CopyField id="account-id" value={awsAccountId || ''} />
              </li>
              <li>
                In the next step, select <strong>Cloud-Meter-policy</strong>.
              </li>
              <li>
                Continue, name the role <strong>Cloud-Meter-role</strong>, and click <strong>Create role</strong>.
              </li>
            </ul>
          </Grid.Col>
        </Form.FormGroup>
      </Form>
    );
  }
}

AccountWizardStepRole.propTypes = {
  getSystemConfig: PropTypes.func,
  awsAccountId: PropTypes.string
};

AccountWizardStepRole.defaultProps = {
  getSystemConfig: helpers.noop,
  awsAccountId: null
};

const mapDispatchToProps = dispatch => ({
  getSystemConfig: () => dispatch(reduxActions.systemConfig.getSystemConfig())
});

const mapStateToProps = state => ({
  awsAccountId: state.accountWizard.configuration[apiTypes.API_SUBMIT_ACCOUNT_AWS_ID]
});

const ConnectedAccountWizardStepRole = connect(
  mapStateToProps,
  mapDispatchToProps
)(AccountWizardStepRole);

export { ConnectedAccountWizardStepRole as default, ConnectedAccountWizardStepRole, AccountWizardStepRole };
