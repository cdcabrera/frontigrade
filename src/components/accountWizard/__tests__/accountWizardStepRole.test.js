import React from 'react';
import { mount } from 'enzyme';
import { AccountWizardStepRole } from '../accountWizardStepRole';

describe('AccountWizardStepRole Component', () => {
  it('should render a wizard policy step', () => {
    const props = {
      configuration: {
        aws_account_id: '123'
      }
    };
    const component = mount(<AccountWizardStepRole {...props} />);

    expect(component.render()).toMatchSnapshot('unconnected');
  });
});
