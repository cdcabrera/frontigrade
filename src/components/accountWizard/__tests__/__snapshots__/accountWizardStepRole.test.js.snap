// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountWizardStepRole Component should render a wizard policy step: unconnected 1`] = `
<form
  class=""
>
  <div
    class="form-group"
  >
    <div
      class="col-sm-12"
    >
      <ul>
        <li>
          In the 
          <strong>
            IAM
          </strong>
           console, create a new role.
        </li>
        <li>
          Choose 
          <strong>
            Another AWS Account
          </strong>
           as the trusted entity type.
        </li>
        <li>
          <p>
            Paste the following content into the 
            <strong>
              Account ID
            </strong>
             field.
          </p>
          <div
            aria-live="polite"
            class="cloudmeter-copy form-group"
          >
            <span
              class="input-group"
            >
              <input
                class="cloudmeter-copy-input false form-control"
                id="account-id"
                readonly=""
                type="text"
                value=""
              />
              <span
                class="input-group-btn"
              >
                <button
                  class="btn btn-default"
                  type="button"
                >
                  Copy
                </button>
              </span>
            </span>
          </div>
        </li>
        <li>
          In the permissions step, attach your new policy. Complete the process to create your new role.
        </li>
      </ul>
    </div>
  </div>
</form>
`;
